buffer/folder.tf

```diff
@@ -13,7 +13,10 @@ resource "google_folder" "pool_parent_folder" {
 # Create one folder for each pool to hold all projects created from that pool
 resource "google_folder" "pool_folders" {
   for_each     = local.create_folders ? toset(var.pool_list) : []
```

Comment:
I think we should remove `toset` so that we can get maintain pool ordering.

Reply:
Had to - for_each only works with unordered sets. I know... bummer. (https://www.terraform.io/docs/configuration/meta-arguments/for_each.html notes that for_each accesses a map or set of strings. Got an error when this was a list)

Reply:
Well I was going to suggest using `count` because that's what we've done historically, but it looks like this isn't recommended. https://www.terraform.io/docs/configuration/meta-arguments/count.html#when-to-use-for_each-instead-of-count @gmalkov fyi (I've certainly seen issues where changing the ordering of a list causes resources to be created & deleted in our terraform).
e.g.
https://github.com/broadinstitute/terraform-ap-modules/blob/9ac6961c86923207615276b4fe586ef9d405b796/buffer/sa.tf#L54-L58.
I'm not wild about using the display name to index the created folders (what if we want to change display names?). It looks like we can use the for_each key on the output resource. https://www.terraform.io/docs/configuration/meta-arguments/for_each.html#referring-to-instances
So maybe some list/map comprehension for the outputs? Not sure exactly what the terraform should look like, but here's a (probably incorrect) stab:
```
output "pool_name_to_folder_id" {
  value = var.enable ? { for p in var.pool_list : p => google_folders.pool_folders[p] } : {}
  ...
```

Reply:
Good idea. Done.